src/QMCWaveFunctions/Fermion/MultiSlaterDetTableMethod.cpp

```diff
@@ -446,7 +446,9 @@ WaveFunctionComponent::PsiValueType MultiSlaterDetTableMethod::ratioGrad(Particl
     new_psi_ratio_to_new_ref_det_ = evalGrad_impl_no_precompute(P, iat, true, dummy);
 
   const int det_id = getDetID(iat);
-  curRatio         = Dets[det_id]->getRefDetRatio() * new_psi_ratio_to_new_ref_det_ / psi_ratio_to_ref_det_;
+  curRatio         = Dets[det_id]->getRefDetRatio();
+  if (curRatio != PsiValueType(0))
```

Comment:
When the proposed configuration is on the nodal surface. `curRatio` goes to zero. This move should be rejected eventually by Monte Carlo. When `Dets[det_id]->getRefDetRatio()` is zero, `new_psi_ratio_to_new_ref_det_` the sum of all the dets with respect to the ref one is `NaN` because it is the sum of all the `1/0`. So keeping `curRatio` zero should be healthy.